src/compiler/compile/render_dom/Block.ts

```diff
@@ -216,215 +225,237 @@ export default class Block {
-						${this.builders.outro}
-					},
-				`);
+				properties.outro = x`function outro(#local) {
+					${this.chunks.outro}
+				}`;
 			}
 		}
 
-		if (this.builders.destroy.is_empty()) {
-			properties.add_line(`d: @noop`);
+		if (this.chunks.destroy.length === 0) {
+			properties.destroy = noop;
 		} else {
-			properties.add_block(deindent`
-				${method_name('d', 'destroy')}(detaching) {
-					${this.builders.destroy}
-				}
-			`);
+			properties.destroy = x`function destroy(detaching) {
+				${this.chunks.destroy}
+			}`;
 		}
 
-		/* eslint-disable @typescript-eslint/indent,indent */
-		return deindent`
-			${this.variables.size > 0 &&
-				`var ${Array.from(this.variables.keys())
-					.map(key => {
-						const init = this.variables.get(key);
-						return init !== undefined ? `${key} = ${init}` : key;
-					})
-					.join(', ')};`}
+		if (!this.renderer.component.compile_options.dev) {
+			// allow shorthand names
+			for (const name in properties) {
+				const property = properties[name];
+				if (property) property.id = null;
+			}
+		}
 
-			${!this.builders.init.is_empty() && this.builders.init}
+		const return_value: any = x`{
+			key: ${properties.key},
+			first: ${properties.first},
+			c: ${properties.create},
+			l: ${properties.claim},
+			h: ${properties.hydrate},
+			m: ${properties.mount},
+			p: ${properties.update},
+			r: ${properties.measure},
+			f: ${properties.fix},
+			a: ${properties.animate},
+			i: ${properties.intro},
+			o: ${properties.outro},
+			d: ${properties.destroy}
+		}`;
+
+		// TODO should code-red do this automatically? probably
+		return_value.properties = return_value.properties.filter(prop => prop.value);
+
+		const body = b`
+			${Array.from(this.variables.values()).map(({ id, init }) => {
+				return init
+					? b`let ${id} = ${init}`
+					: b`let ${id}`;
+			})}
+
+			${this.chunks.init}
 
 			${dev
-				? deindent`
-					const block = {
-						${properties}
-					};
-					@dispatch_dev("SvelteRegisterBlock", { block, id: ${this.name || 'create_fragment'}.name, type: "${this.type}", source: ${stringify(this.comment || '', { only_escape_at_symbol: true })}, ctx });
+				? b`
+					const block = ${return_value};
+					@dispatch_dev("SvelteRegisterBlock", { block, id: ${this.name || 'create_fragment'}.name, type: "${this.type}", source: "${this.comment ? this.comment.replace(/"/g, '\\"') : ''}", ctx: #ctx });
 					return block;`
-				: deindent`
-					return {
-						${properties}
-					};`
+				: b`
+					return ${return_value};`
 			}
-		`.replace(/([^{])(#+)(\w*)/g, (_match: string, pre: string, sigil: string, name: string) => {
-			return pre + (sigil === '#' ? this.alias(name) : sigil.slice(1) + name);
-		});
-		/* eslint-enable @typescript-eslint/indent,indent */
+		`;
+
+		return body;
+	}
+
+	render() {
+		const key = this.key && this.get_unique_name('key');
+
+		const args: any[] = [x`#ctx`];
+
+		if (key) args.unshift(key);
+
+		// TODO include this.comment
```

Comment:
Is this for the comments that tell you what a particular fragment is for? Looks like this still need to be re-implemented.

Reply:
yep — will do after this PR